File: server/server/Models/Entities/Book.cs
namespace server.Models.Entities
{
    public class Book
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public string Description { get; set; }
        public required string Author { get; set; }
        public required string CoverImageUrl { get; set; }
        public required string Category { get; set; }
        public string Publisher { get; set; }
        public required string Isbn { get; set; }
        public string PublicationDate { get; set; }
    }
}

File: client/src/app/models/book.ts
export interface Book {
  id?: string; 
  title: string;
  description: string;
  author: string;
  isbn: string;
  publicationDate: Date;
  publisher: string;
  coverImageUrl?: string;
  category: string;
}

export interface AddBookDto {
  title: string;
  description: string;
  author: string;
  isbn: string;
  publicationDate: Date;
  publisher: string;
  coverImageUrl?: string;
  category: string;
}

export interface UpdateBookDto {
  title: string;
  description: string;
  author: string;
  isbn: string;
  publicationDate: Date;
  publisher: string;
  coverImageUrl?: string;
  category: string;
}